internals/overlord/logstate/loki/loki.go

```diff
@@ -44,45 +43,42 @@ type Client struct {
 
 	// To store log entries, keep a buffer of size 2*MaxRequestEntries with a
 	// sliding window 'entries' of size MaxRequestEntries
-	buffer  []lokiEntryWithService
-	entries []lokiEntryWithService
+	buffer  []entryWithService
+	entries []entryWithService
 
 	// store the custom labels for each service
 	labels map[string]json.RawMessage
 }
 
-func NewClient(target *plan.LogTarget) *Client {
-	return NewClientWithOptions(target, &ClientOptions{})
-}
-
-// ClientOptions allows overriding default parameters (e.g. for testing)
-type ClientOptions struct {
-	RequestTimeout    time.Duration
-	MaxRequestEntries int
-}
-
-func NewClientWithOptions(target *plan.LogTarget, options *ClientOptions) *Client {
-	options = fillDefaultOptions(options)
+func NewClient(target *plan.LogTarget, options *ClientOptions) *Client {
```

Comment:
Per comment on `opentelemetry.go`, I'd rather avoid the dependency on the plan package, and have all the options in `ClientOptions` (like we had before with `opentelemetry.NewClient`). That seems cleaner to me -- fewer dependencies, as these packages don't need to know about the Pebble plan.